controlplane/kubeadm/internal/controllers/status.go

```diff
@@ -577,30 +568,111 @@ func setAvailableCondition(_ context.Context, kcp *controlplanev1.KubeadmControl
 			controlplanev1.KubeadmControlPlaneMachineEtcdMemberHealthyV1Beta2Condition,
 			controlplanev1.KubeadmControlPlaneMachineEtcdPodHealthyV1Beta2Condition) {
 			k8sControlPlaneNotHealthy++
+		} else {
+			k8sControlPlaneNotHealthyButNotReportedYet++
 		}
 	}
 
+	// Maps machines to members
+	memberToMachineMap := map[string]*clusterv1.Machine{}
+	provisioningMachines := []*clusterv1.Machine{}
+	for _, machine := range machines {
+		if machine.Status.NodeRef == nil {
+			provisioningMachines = append(provisioningMachines, machine)
+			continue
+		}
+		for _, member := range etcdMembers {
+			if machine.Status.NodeRef.Name == member.Name {
```

Comment:
So we assume the Node name is identical to the member name, right?

Is this an assumption that we already make in other places or is this new? (and I mean places that already existed with the last minor release)